frontend-opal/src/thunks/searchResult.js

```diff
@@ -90,24 +53,19 @@ function mapTitle(data) {
 }
 
 function search(input) {
-  const initRes = {
-    input,
-    results: []
-  };
   return async dispatch => {
+    let initRes = {
+      results: []
+    };
     if (input !== '') {
       const searchRes = await searchData(input);
-      const res = mapTitle(searchRes);
-
-      dispatch(
-        actions.search({
-          input,
-          results: res
-        })
-      );
-    } else {
-      dispatch(actions.search(initRes));
+      initRes = mapTitle(searchRes);
     }
+    dispatch(
+      actions.search({
+        results: initRes
```

Comment:
@Yuan-EPAM are you sure to dispatch this action payload:
```javascript
{
  results: {
     results: [...]
  }
}
```

Reply:
also don't call it initRes, it is the result :)

Reply:
Ops, strange thing. Actually in my local file, the dispatch function was `dispatch(actions.search(initRes)`. 

And how about the name of  `data`~